Sources/Datadog/RUM/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -36,31 +48,55 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
     // MARK: - RUMScope
 
     func process(command: RUMCommand, context: DatadogContext, writer: Writer) -> Bool {
-        if sessionScope == nil {
+        if sessionScopes.isEmpty && !applicationActive {
             startInitialSession(on: command, context: context, writer: writer)
         }
 
-        if let currentSession = sessionScope {
-            sessionScope = sessionScope?.scope(byPropagating: command, context: context, writer: writer)
+        if activeSession == nil && command.isUserInteraction {
+            // No active sessions, start a new one
+            startNewSession(on: command, context: context, writer: writer)
+        }
+
+        if command is RUMStopSessionCommand {
+            // Reach in and grab the last active view
+            lastActiveView = activeSession?.viewScopes.first(where: { $0.isActiveView })
+        }
+
+        // Can't use scope(byPropagating:context:writer) because of the extra step in looking for sessions
+        // that need a refresh
+        sessionScopes = sessionScopes.compactMap({ scope in
+            if scope.process(command: command, context: context, writer: writer) {
+                // Returned true, keep the scope around, it still has work to do.
+                return scope
+            }
 
-            if sessionScope == nil { // if session expired
-                refresh(expiredSession: currentSession, on: command, context: context, writer: writer)
+            if scope.isActive {
+                // False, but still active means we timed out or expired, refresh the session
```

Comment:
From the comment it feels like `isSessionActive` should be checked here as well?

I don't see this param used elsewhere.

Reply:
I don't get what `False` is referring to here, since we are inside of `isActive`, which is true

Reply:
Ah, It's the return value of `scope.process` If it returned `true` return the scope, False, we check isActive. I'll change the comment to make it clearer.